bin/inference/pycbc_inference_plot_inj_recovery

```diff
@@ -0,0 +1,143 @@
+    zlabel = None
+
+    # loop over input files
+    logging.info("Reading %s values", opts.z_arg)
+    for i, input_fp in enumerate(fp):
+
+        # get z-axis values and label
+        likelihood_stats = input_fp.read_likelihood_stats(
+             thin_start=opts.thin_start, thin_end=opts.thin_end,
+             thin_interval=opts.thin_interval, iteration=opts.iteration)
+        vals, zlabel = option_utils.get_zvalues(input_fp, opts.z_arg,
+                                                likelihood_stats)
+        zvals.append(numpy.median(vals))
+
+        # update range of colorbar
+        min_zval = vals.min() if i == 0 else min(min_zval, vals.min())
+        max_zval = vals.max() if i == 0 else max(max_zval, vals.max())
+
+    # create colormap
+    cmap = cm.get_cmap(opts.scatter_cmap)
+    vmin = opts.vmin if opts.vmin else min_zval
+    vmax = opts.vmax if opts.vmax else max_zval
+    norm = mpl.colors.Normalize(vmin, vmax)
+
+# loop over input files and its samples
+logging.info("Plotting")
+for i, (input_file, input_fp, input_samples) in enumerate(zip(opts.input_file,
+                                                              fp, samples)):
+
+    # read injections from HDF input file
+    injs = inject.InjectionSet(input_file, hdf_group=opts.injection_hdf_group)
+
+    # check if need extra parameters than parameters stored in injection file
+    _, ts = transforms.get_common_cbc_transforms(opts.parameters,
+                                                 injs.table.fieldnames)
+
+    # add parameters not included in injection file
+    inj_parameters = transforms.apply_transforms(injs.table, ts)
+
+    # get paramter values
+    sampled_vals = input_samples[parameter]
+    injected_vals = inj_parameters[parameter]
+
+    # compute quantiles of sampled results
+    quantiles = numpy.array([numpy.percentile(sampled_vals, 100 * q)
+                             for q in opts.quantiles])
+
+    # get median and lowest and highest quntiles for plotting
+    med = numpy.median(sampled_vals) - injected_vals
+    high = quantiles.max() - injected_vals
+    low = quantiles.min() - injected_vals
+
+    # get color
+    if opts.z_arg:
+        color = cmap(norm(zvals[i]))
+    else:
+        color = "black"
+
+    # plot a point for each injection
+    if len(injected_vals) > 1:
+        logging.warn("Mrne than one injection in file %s", input_file)
+    ax.errorbar([injected_vals],
+                [med],
+                yerr=[[med - low], [high - med]],
+                ecolor=color, linestyle="None", zorder=10)
+
+# create a colorbar
+if opts.z_arg:
+    cax, _ = cbar.make_axes(ax)
+    cb2 = cbar.ColorbarBase(cax, cmap=cmap, norm=norm)
+    cb2.set_label(r"Recovered Median " + zlabel)
+
+# set labels
+ax.set_ylabel(r"Recovered " + label + r"- Injected " + label)
+ax.set_xlabel(r"Injected " + r"{}".format(label))
+
+# add grid to plot
+ax.grid()
+
+# add 1:1 line to plot
+ax.axhline(0, linestyle="dashed", color="gray", zorder=9)
```

Comment:
The y-value here is being set to 0 but the gray dashed line in the [plot](https://sugwg-jobs.phy.syr.edu/~cbiwer/projects/inference_write_inj/plots/test2.png) is not at 0. What am I missing?

Reply:
Commit was pushed later to change from ratio to absolute difference. Therefore change from 1 to 0.